analyzers/src/SonarAnalyzer.CSharp/Rules/MesageTemplates/MessageTemplateAnalyzer.cs

```diff
@@ -0,0 +1,109 @@
+{
+    private static readonly HashSet<string> MicrosoftExtensionsLoggingMethods =
+        [
+            "Log",
+            "LogCritical",
+            "LogDebug",
+            "LogError",
+            "LogInformation",
+            "LogTrace",
+            "LogWarning",
+        ];
+
+    private static readonly HashSet<string> SerilogMethods =
+        [
+            "Debug",
+            "Error",
+            "Information",
+            "Fatal",
+            "Warning",
+            "Write",
+            "Verbose",
+        ];
+
+    private static readonly HashSet<string> NLogLoggerMethods =
+        [
+            "Debug",
+            "Error",
+            "Fatal",
+            "Info",
+            "Log",
+            "Trace",
+            "Warn",
+            "ConditionalTrace",
+            "ConditionalDebug",
+        ];
+
+    private static readonly HashSet<string> NLogLoggerBaseMethods = ["Log"];
+
+    private static readonly ImmutableHashSet<IMessageTemplateCheck> Checks = ImmutableHashSet.Create<IMessageTemplateCheck>(
+               new NamedPlaceholdersShouldBeUnique());
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+        Checks.Select(x => x.Rule).ToImmutableArray();
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+        {
+            var enabledChecks = Checks.Where(x => x.Rule.IsEnabled(c)).ToArray();
+            if (enabledChecks.Length > 0
+                && c.Node is InvocationExpressionSyntax invocation
+                && TemplateArgument(invocation, c.SemanticModel) is { } argument
+                && MessageTemplates.Parse(argument.Expression.ToString()) is { Success: true } result)
+            {
+                foreach (var check in enabledChecks)
+                {
+                    check.Execute(c, invocation, argument, result.Placeholders);
+                }
+            }
+        },
+        SyntaxKind.InvocationExpression);
+
+    private static ArgumentSyntax TemplateArgument(InvocationExpressionSyntax invocation, SemanticModel model) =>
+        TemplateArgument(invocation, model, KnownType.Microsoft_Extensions_Logging_LoggerExtensions, MicrosoftExtensionsLoggingMethods, "message")
+        ?? TemplateArgument(invocation, model, KnownType.Serilog_Log, SerilogMethods, "messageTemplate")
+        ?? TemplateArgument(invocation, model, KnownType.Serilog_ILogger, SerilogMethods, "messageTemplate")
+        ?? TemplateArgument(invocation, model, KnownType.NLog_ILogger, NLogLoggerMethods, "message", exactMatch: false)
+        ?? TemplateArgument(invocation, model, KnownType.NLog_ILoggerBase, NLogLoggerBaseMethods, "message", exactMatch: false);
+
+    private static ArgumentSyntax TemplateArgument(InvocationExpressionSyntax invocation, SemanticModel model, KnownType type, HashSet<string> methods, string template, bool exactMatch = true) =>
```

Comment:
You can use the most generic type.
```suggestion
    private static ArgumentSyntax TemplateArgument(InvocationExpressionSyntax invocation, SemanticModel model, KnownType type, ICollection<string> methods, string template, bool exactMatch = true) =>
```